### PR TITLE
Pbi 1 convert jet uml class diagram files to code

### PR DESCRIPTION
**Pull request by KenKomKom**
Implemented functionality to parse a class diagram from a JSON string, extracting class names, attributes, and relationships. This allows the translation from JSON to premade classes in the application

### CODE REVIEW

**Comment by DaWanAnOnli** [comment]
- Next time, just make sure that commits with the 'implement' message will actually pass all tests. As from here, we can't tell which tests are passed for each commit. Therefore if an 'implement' commit is still [RED], we don't know whether the failed tests are related to the  tests this commit is trying to pass or not.

File: app/main.py [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+async def download_file(request : DownloadRequest, background_tasks: BackgroundTasks):
+    file = request.filename+".py"
+
+    if '/' in request.filename or '\\' in request.filename:
+        raise HTTPException(status_code=400, detail="/ not allowed")
+
+    if os.path.exists(file):
+        raise HTTPException(status_code=400, detail="The file already exists.")
+
+    with open(file, "w") as f:
```
Comment (MightyZanark) [inline app/main.py:24]: for opening files in async functions, use `aiofiles` instead so it does not block the thread, might also want to use the `tempfile` module instead of creating and deleting it manually

**Comment by MightyZanark** [comment]
Other than the ones commented, might want to use the linter and formatter first as well

File: tests/test_parse_class.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+                "type": "ClassNode"
+                }
+            ]
+        }"""
+        parser = ParseJsonToObject(data)
+        result = parser.parse_classes()
+        self.assertEqual(result, parser._ParseJsonToObject__classes)
+        self.assertEqual(parser._ParseJsonToObject__json, json.loads(data))
+        self.assertEqual(len(parser._ParseJsonToObject__json["nodes"]), 1)
+        self.assertEqual(parser._ParseJsonToObject__json["nodes"][0]["name"], "Pengelola")
```
Comment (NtapSlur) [inline tests/test_parse_class.py:102]: You can add test to verify what the program will do if it detects a missing class name since it can cause an error if it's proccessed

Comment (NtapSlur) [reply tests/test_parse_class.py:102]: Here are some other notes that can be taken:
- It is better to change the name of the file **parse_json_to_object.py** since there will also be a parser for the Sequence Diagram. It will be hard to identify which parser is for which diagram if it is not specified in the file name for the UML Diagram type. 
- It is also better if the test case includes the other edge case of the program like duplicate class name or attribute

Other than that, the rest is already good! 👍

File: app/main.py (hunk 1 above)
Comment (KenKomKom) [reply app/main.py:24]: By aiofiles, you mean these right? [aiofiles](https://pypi.org/project/aiofiles/) [https://docs.python.org/3/library/tempfile.html](https://docs.python.org/3/library/tempfile.html).
I'll fix it for the next pull request

File: tests/test_parse_class.py (hunk 2 above)
Comment (KingOnyxx) [reply tests/test_parse_class.py:102]: Thanks for noticing! I'll try to implement it.
